src/main/java/dev/imabad/mceventsuite/spigot/modules/eventblocker/EventBlockListener.java

```diff
@@ -0,0 +1,116 @@
+package dev.imabad.mceventsuite.spigot.modules.eventblocker;
+
+import com.google.gson.JsonObject;
+import dev.imabad.mceventsuite.spigot.config.SpigotConfig;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.*;
+import org.bukkit.event.entity.*;
+import org.bukkit.event.hanging.HangingPlaceEvent;
+import org.bukkit.event.inventory.*;
+import org.bukkit.event.player.*;
+import org.bukkit.event.raid.RaidTriggerEvent;
+import org.bukkit.event.world.StructureGrowEvent;
+
+import java.io.IOException;
+
+public class EventBlockListener implements Listener {
+
+    SpigotConfig configFile;
+    JsonObject config;
+
+    public EventBlockListener() {
+        try {
+            configFile = new SpigotConfig("config");
+            config = configFile.getJson();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @EventHandler
+    public void onEvent(Event e) {
+        //If the configuration has the event
+        if (config.has("eventblocker." + e.getClass().getSimpleName().toLowerCase())) {
+            //If the event is a redstone event, cancel it
+            if (!config.get("eventblocker." + e.getClass().getSimpleName().toLowerCase()).getAsBoolean() && BlockRedstoneEvent.class.isAssignableFrom(e.getClass())) {
+                ((BlockRedstoneEvent) e).setNewCurrent(((BlockRedstoneEvent) e).getOldCurrent());
+            }
+
+            //If the event is cancellable, check to see if you can cancel it.
+            else if (!config.get("eventblocker." + e.getClass().getSimpleName().toLowerCase()).getAsBoolean() && Cancellable.class.isAssignableFrom(e.getClass())) {
+
+                //If there's a player involved, set the player
+                Player player = checkPlayer(e);
+
+                // If there is no player, or the player does not have permissions to allow the event, cancel te event.
+                if (player == null || (!player.isOp() || !player.hasPermission("eventblocker." + e.getClass().getSimpleName().toLowerCase()))) ((Cancellable) e).setCancelled(true);
+
+            }
+        }
+    }
+
+    private Player checkPlayer(Event e) {
+        //TODO: Find a way to make this a little cleaner
+        Player player = null;
+        if (BlockBreakEvent.class.isAssignableFrom(e.getClass())) {
+            player = ((BlockBreakEvent) e).getPlayer();
```

Comment:
Can we try to find a way of using reflection to see if a getPlayer method exists? If you get stuck on that let me know and I can play.

Reply:
I couldn't find a way to get it to work without getting the server to crash :/